applications/StructuralMechanicsApplication/custom_constitutive/constitutive_laws_integrators/generic_constitutive_law_integrator_damage.h

```diff
@@ -0,0 +1,288 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Alejandro Cornejo & Lucia Barbu
+//
+
+#if !defined(KRATOS_GENERIC_CONSTITUTIVE_LAW_INTEGRATOR_DAMAGE_H_INCLUDED)
+#define  KRATOS_GENERIC_CONSTITUTIVE_LAW_INTEGRATOR_DAMAGE_H_INCLUDED
+
+// System includes
+#include <string>
+#include <iostream>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/serializer.h"
+#include "includes/properties.h"
+#include "utilities/math_utils.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class GenericConstitutiveLawIntegratorDamage
+ * @ingroup StructuralMechanicsApplication
+ * @brief: This object integrates the predictive stress using the isotropic damage theory by means of 
+ * linear/exponential softening.
+ * @details
+ * @tparam TYieldSurfaceType
+ * @author Alejandro Cornejo & Lucia Barbu
+ */
+template <class TYieldSurfaceType> //, class TVoigtSize>
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) GenericConstitutiveLawIntegratorDamage
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+
+    /// The type of yield surface
+    typedef TYieldSurfaceType YieldSurfaceType;
+
+    /// The type of plastic potential 
+    typedef typename YieldSurfaceType::PlasticPotentialType PlasticPotentialType;
+
+    /// Counted pointer of GenericConstitutiveLawIntegratorDamage
+    KRATOS_CLASS_POINTER_DEFINITION(GenericConstitutiveLawIntegratorDamage);
+
+    /// Initialization constructor
+    GenericConstitutiveLawIntegratorDamage()
+    {
+    }
+
+    /// Copy constructor
```

Comment:
I would recommend to not implement copy & assignment constructor if they do the same as the ones created by the compiler

Reply:
this still applies